src/Pester.Runtime.ps1

```diff
@@ -2604,3 +2608,20 @@ function Add-MissingContainerParameters ($RootBlock, $Container, $CallingFunctio
 
     $RootBlock.FrameworkData.MissingParametersProcessed = $true
 }
+
+function Assert-BoundScriptBlockInput {
+    param(
+        [Parameter(Mandatory = $true)]
+        [ScriptBlock] $ScriptBlock
+    )
+    $internalSessionState = $script:ScriptBlockSessionStateInternalProperty.GetValue($ScriptBlock, $null)
+    if ($null -eq $internalSessionState) {
+        $maxLength = 250
+        $prettySb = (Format-Nicely2 $ScriptBlock) -replace '\s{2,}', ' '
+        if ($prettySb.Length -gt $maxLength) {
+            $prettySb = $prettySb.Remove($maxLength) + "..."
+        }
+
+        throw [System.ArgumentException]::new("Unbound scriptblock '$prettySb' is not allowed. See https://github.com/pester/Pester/issues/2411.")
```

Comment:
```suggestion
        throw [System.ArgumentException]::new("Unbound scriptblock is not allowed to be used, because it would run inside of Pester session state, which would produce unexpected results. See https://github.com/pester/Pester/issues/2411 for more details and workarounds. ScriptBlock:  '$prettySb'") 
```

how about this? I think explaining why the scriptblock is not allowed first and then showing it would make it easier to see what the problem is, rather than saying 

The scriptblock 
then the loooooooooooooong scriptblock
then the loooooooooooooong scriptblock
then the loooooooooooooong scriptblock
then the loooooooooooooong scriptblock
then the loooooooooooooong scriptblock
and then the actual error.

Reply:
Good point! Fixed